src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineCloud.java

```diff
@@ -365,39 +375,46 @@ private synchronized Integer availableNodeCapacity() throws IOException {
   @Override
   public boolean canProvision(Label label) {
     try {
-      getInstanceConfig(label);
+      getInstanceConfigurations(label);
       return true;
     } catch (NoConfigurationException nce) {
       return false;
     }
   }
 
-  /** Gets {@link InstanceConfiguration} that has the matching {@link Label}. */
-  public InstanceConfiguration getInstanceConfig(Label label) throws NoConfigurationException {
+  /** Gets all instances of {@link InstanceConfiguration} that has the matching {@link Label}. */
+  public List<InstanceConfiguration> getInstanceConfigurations(Label label)
+      throws NoConfigurationException {
```

Comment:
This method should not throw exception at all..
But I did not wanted to fix it here..
Another PR should be done to clean it up..

Reply:
Hmm do you mean in the case where we just have a cloud and no configurations (which I agree is fine), so we shouldn't be throwing exceptions?

Reply:
yes.. we should return empty list..